### PR TITLE
See the creation of analysis for an area

### PR DESCRIPTION
**Pull request by fertandil87**
https://globalfishingwatch.atlassian.net/browse/MAP-1218

https://github.com/GlobalFishingWatch/frontend/assets/1672451/e3c069c5-6331-44ae-b033-cbfd40f9706a

image

### CODE REVIEW

File: apps/fishing-map-e2e/src/e2e/map/analysis.cy.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  })
+  beforeEach(() => {
+    disablePopups()
+    cy.visit('/')
+    waitForSidebarLoaded()
+    waitForMapLoadTiles()
+    scrollSidebar('bottom', 1000)
+    cy.getBySel(SIDEBAR_TOGGLE_EEZ).click()
+    cy.getBySel('map-search-button').click()
+    // I need to add a delay because it doesnt work propetly the autocomplete if we type so fast
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/e2e/map/analysis.cy.ts:24]: That's probable caused by the debounce that was set to the input text change event.

File: apps/fishing-map-e2e/src/e2e/map/analysis.cy.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    getSidebar().findBySelLike('report-title').contains(SEARCH_EEZ_FULL_NAME)
+    cy.getBySel('source-tags').findBySelLike('source-tag-item').contains('AIS')
+
+    // Path tag is a needed element that should exist to draw the charts
+    cy.getBySel('report-activity-evolution').find('path').should('exist')
+    cy.getBySel('report-vessels-graph').find('path').should('exist')
+    cy.getBySel('report-vessels-table').findBySelLike('vessel').should('exist')
+  })
+
+  it('Should create an analysis for an EEZ area', () => {
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/e2e/map/analysis.cy.ts:42]: Why do we have two tests with the same name?  👉  https://github.com/GlobalFishingWatch/frontend/pull/2153/files#diff-303739775a7f1ddc13229986b1073f0dc8fb4c86924ea260a1fbd091d740c127R32

File: apps/fishing-map-e2e/src/support/app.po.ts [hunk 3]
```diff
@@ -50,3 +49,6 @@ export const switchLanguage = (language: string) => {
     cy.reload()
   }
 }
+
+export const scrollSidebar = (position, duration = 0) =>
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/support/app.po.ts:53]: 👍

File: apps/fishing-map/features/workspace/shared/DatasetSourceField.tsx [hunk 4]
```diff
@@ -37,9 +37,10 @@ function DatasetFilterSource({ dataview, hideColor }: DatasetFilterSourceProps)
   }
 
   return (
-    <div className={styles.filter} data-test="activity-filters-tags">
+    <div className={styles.filter} data-test="source-tags">
       <label>{t('layer.source', 'Sources')}</label>
       <TagList
+        testId="source-tag-item"
```
Comment (rdgfuentes) [inline apps/fishing-map/features/workspace/shared/DatasetSourceField.tsx:43]: Why did we changed these `data-test` attributes? This change is going to make the vms tests fail: https://github.com/GlobalFishingWatch/frontend/blob/ee8ab566b5e3141d4396bb88bc20becef285486a/apps/fishing-map-e2e/src/e2e/fishing-activity/vms.cy.ts#L10-L21

File: libs/ui-components/src/tag-list/TagList.tsx [hunk 5]
```diff
@@ -27,7 +27,7 @@ export function TagList(props: TagListProps) {
   return (
     <ul className={cx(styles.tagList, className)}>
       {tags.map((tag, index) => (
-        <li key={tag.id || index} data-test={`${testId}-${tag.id}`}>
+        <li key={tag.id || index} {...(testId && { 'data-test': `${testId}-${tag.id}` })}>
```
Comment (rdgfuentes) [inline libs/ui-components/src/tag-list/TagList.tsx:30]: Good catch!

File: apps/fishing-map-e2e/src/e2e/map/analysis.cy.ts (hunk 1 above)
Comment (fertandil87) [reply apps/fishing-map-e2e/src/e2e/map/analysis.cy.ts:24]: I don't know but I spent lot of time trying to understand why doing it fast was not working

**Pull request merged**